plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -257,6 +257,11 @@ export class GithubEntityProvider implements EntityProvider, EventSubscriber {
     const catalogFile = this.config.catalogPath.startsWith('/')
       ? this.config.catalogPath.substring(1)
       : this.config.catalogPath;
+
+    if (branch.includes('/')) {
+      // Fixes https://github.com/backstage/backstage/issues/16403
+      return `${repository.url}/blob/${branch}?path=${catalogFile}`;
+    }
     return `${repository.url}/blob/${branch}/${catalogFile}`;
```

Comment:
```suggestion
    return `${repository.url}/blob/${branch}/?path=${catalogFile}`;
```
A included the `if` and the return trying to "fix what is not working, without changing what is working".

What are the consequences if we return only this more generic url type?

Reply:
That's the tricky thing; we're not entirely sure :) There's a fair amount of direct usage of `git-url-parse`; does that library support that format directly?

One more place where this comes into play is [the GithubIntegration class](https://github.com/backstage/backstage/blob/2486cae0bd70cf379030495bded02d88e70347a8/packages/integration/src/github/GithubIntegration.ts#L29) whose tests should probably be amended to run all of the existing tests with the new format too. Especially `resolveUrl` is used to find URLs relative to a base.

So indeed let's keep the `if` to be sure for now, and then amend the tests as suggested, and if you get the chance to report back on whether raw `git-url-parse` seems to handle the format nicely I'd be happy about that too.

Reply:
Hey @freben! The library doesn't support this format yet. That's why I had to implement [this behavior extension](https://github.com/backstage/backstage/pull/16407/files#diff-8f657a161a6dc0770a5d2d7fb597a7c48121a82a9b8435997a4429a620dee63fR42-R51) in the client.
What do you think of wrapping the library with this behavior in a module and importing the module instead of the library throughout all the codebase?

About vendors: I known GitHub supports it but I don't known if the format is universal between gitlab, bitbucket, etc...

Thank you for the advice. I will amend the tests later this week. Sorry for taking this long to answer.

Reply:
Can you check [this commit](https://github.com/backstage/backstage/pull/16407/commits/1555455005323ef1dd5fb10e2a501483ddcbf4d4)? I am not sure of the behavior I should expect from the `GithubIntegration` class regarding this format. Should it ignore (and just return it as is) or should it resolve to a fully qualified git url?
I think I don't known what role `GithubIntegration` plays on the overall ingestion from github.